tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -3497,6 +3509,9 @@ uint32_t CGMSHLSLRuntime::AddUAVSRV(VarDecl *decl,
   if (decl->hasAttr<HLSLGloballyCoherentAttr>()) {
     hlslRes->SetGloballyCoherent(true);
   }
+  if (decl->hasAttr<HLSLReorderCoherentAttr>()) {
+    hlslRes->SetReorderCoherent(true);
+  }
```

Comment:
nit: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
```suggestion
  if (decl->hasAttr<HLSLReorderCoherentAttr>())
    hlslRes->SetReorderCoherent(true);
```